### DISCUSSION TITLE
Unofficial native Mouse & Keyboard supported games

### DISCUSSION THREAD

**Discussion by redphx** [redphx/better-xcloud#574] [General]
> [!WARNING]
> I can't add native MKB support if the game doesn't support it. Test the game yourself. Ask me = delete.

> [!NOTE] 
> The `Force native M&KB for these games`'s game list can be updated retroactively without having to update the script. You may need to refresh the page twice to see the updated list.

## How to test:

1. Uninstall Better xCloud script (you won't lose script settings)
2. Create a new script in Tampermonkey (code below)
3. Reinstall Bx script
4. Test added game(s)
5. Report the results

```javascript
// ==UserScript==
// @name         Better xCloud - Custom flags
// @namespace    https://github.com/redphx
// @version      1.0.0
// @description  Customize Better xCloud script
// @author       redphx
// @license      MIT
// @match        https://www.xbox.com/*/play*
// @run-at       document-start
// @grant        none
// ==/UserScript==
'use strict';

/*
Make sure this script is being loaded before the Better xCloud script.

How to:
1. Uninstall Better xCloud script.
2. Install this script.
3. Reinstall Better xCloud script. All your settings are still there.
*/

// Change this to `false` if you want to temporary disable the script
const enabled = true;

enabled && (window.BX_FLAGS = {
    /*
    Add titleId of the game(s) you want to add here.
    Keep in mind: this method only works with some games.

    Example:
        - Flight Simulator has this link: /play/games/microsoft-flight-simulator-standard-40th-anniversa/9PMQDM08SNK9
        - That means its titleId is "9PMQDM08SNK9"
        - So it becomes:
            ForceNativeMkbTitles: [
                "9NJDD0JGPP2Q", // AoE II

            ],
    */
    ForceNativeMkbTitles: [
        "9P2VGCTBMM52", // MSFS2024

        "9NJDD0JGPP2Q", // Age of Empires II
        "9NS4Q51TVR51", // All You Need Is Help
        "9NJWFF2KHL6H", // Ben 10
        "9N7GJJ2WTPBZ", // Football Manager Console 2024
        "9NN3HCKW5TPC", // Gears Tactics
        "9MWHMJ0SRBXV", // Jurassic World Evolution 2
        "9NBH78WD7XPQ", // The Case of the Golden Idol
    ],
});
```

After following these steps, if you want to add/remove games you only need to edit the script at step 2.

## [Supported games](https://github.com/redphx/better-xcloud/blob/gh-pages/native-mkb/ids.json):
- 7 Days to Die
- ANIMAL WELL
- Assassin's Creed Mirage
- Assassin's Creed Valhalla
- Cyberpunk 2077
- DayZ
- Dungeons Of Hinterberg
- Fear the Spotlight
- Forza Motorsport
- Gears 5
- Harry Potter: Quidditch Champions
- Hogwarts Legacy
- Hogwarts Legacy (Xbox One)
- Immortals Fenyx Rising
- Metro Exodus
- Microsoft Flight Simulator 2020
- Palworld
- Prince of Persia The Lost Crown
- STAR WARS Jedi: Fallen Order

## Bugged:  
- JURASSIC WORLD EVOLUTION 2

## Unsupported:
- [MSFS 2024](https://github.com/redphx/better-xcloud/discussions/574#discussioncomment-11414322)
- Persona 3 Reload

## Need to be tested:
- Age of Empires II (https://www.xbox.com/play/games/9NJDD0JGPP2Q)
- All You Need Is Help (https://www.xbox.com/play/games/9NS4Q51TVR51)
- Ben 10 (https://www.xbox.com/play/games/9NJWFF2KHL6H)
- FF XIV (many editions, get ID manually)
- Football Manager Console 2024 (https://www.xbox.com/play/games/9N7GJJ2WTPBZ)
- Gears Tactics (https://www.xbox.com/play/games/9NN3HCKW5TPC)
- Jurassic World Evolution 2 (https://www.xbox.com/play/games/9MWHMJ0SRBXV)
- The Case of the Golden Idol (https://www.xbox.com/play/games/9NBH78WD7XPQ)

There are [other games](https://www.purexbox.com/guides/all-xbox-games-with-mouse-and-keyboard-support) that support MKB. You can add any game you want.

**Comment by ricardo404** [comment]
Hi Red, I tested it on MSFS 2024 and despite the xcloud layout I interacted as if the game had keyboard and mouse support, there is no effect in the game. No mouse or keyboard control. I imagine it's a platform issue, right? I haven't tested it on 2020 yet, both are my favorite games and simulators, my use on Xcloud is basically around the two simulators.

**Comment by eduxz122** [comment]
hello, assasin's creed valhalla is working pretty well

**Comment by pdario** [comment]
I'm trying MSFS 2024, but nothing happens; this list is empty, is this correct?

![image](https://github.com/user-attachments/assets/95f256f4-d9d7-4be8-94a0-f97b133d94f5)

**Reply by ricardo404** [reply]
I believe the feature is turned off on Xcloud's side. Out of Redphx's control I think.

**Reply by Stuarta99** [reply]
having the same issue so guess nothing can be done?

**Comment by AlvaroG117** [comment]
hi, i've tested cyberpunk 2077, it's working

**Reply by redphx** [reply]
added, thanks.

**Comment by mati123412** [comment]


**Reply by oasted** [reply]
Yes please

**Comment by iosifc23** [comment]
Tried immortals-fenyx-rising/C07KJZRH0L7S . Native Kb&M is working .

**Reply by redphx** [reply]
added, thanks

**Reply by PlutoPhantomWolf** [reply]
man i tried immortals fenyx rising, I am stuck on the home screen where I can't press Space to Continue. 

its not working for me
please help.

**Comment by Klaedy** [comment]
2 days playing Dungeons of Hinterberg and everything OK

**Reply by redphx** [reply]
added, thanks

**Comment by iosifc23** [comment]
Tested star-wars-jedi-fallen-order/C2CSDTSCBZ0C . Is working .

**Reply by redphx** [reply]
added, thanks

**Comment by ricardo404** [comment]
People who fly in FS2020: how do I get around a BUG that happens, if you start 2020 with the "force mouse and keyboard" active, the controls menu inside the simulator turns gray and the controls in the plane are bugged. Is there a trick to make it work without this problem?

**Reply by AnonCYTO** [reply]
hey i face that bug too from time to time. i just enter and exit a free flight then its not gray anymore. would love to know what's going on behind the scenes from someone who knows better.

**Comment by LukyHale** [comment]
How do I test it on Dayz, is there a video teaching it to a layman?

**Reply by nomadliam** [reply]
BSR9NLHVF1KL is the dayz code for the script.  as far as I can tell it works on microsoft edge only after ptessing a on controller to enter the menu

**Comment by Sparkyz258** [comment]
the game JURASSIC WORLD EVOLUTION 2 with mouse and keyboard is bugged, as the mouse cursor is not appearing.

**Comment by ifoodplayer** [comment]
astroneer/9NBLGGH43KZB game is working but every time you open it you need to press A to continue only once per opening

**Comment by joihui** [comment]
I already tested palword (9NKV34XDW014) and forza motorsport (9PLKVSWR299F) working

**Comment by nomadliam** [comment]
BSR9NLHVF1KL - DayZ - Keyboard & Mouse working fine on edge but not on brave browser. Took me a while to realise that brave isnt allowing the mouse to be captured it seems.

**Reply by oZifer** [reply]
how did u do that

**Reply by oZifer** [reply]
tried on chrome and edge and it dont worked for me can you send a video or a print of how did u make it

**Reply by nomadliam** [reply]
> how did u do that

You dont need this test script for Day z, i just used it to test on brave as it wouldn't work with the mouse at all in brave. 

How to play on edge with kbm - Install better xcloud extension on microsoft edge via the official website (tapermonkey & better xcloud script) open up the bettwr xcloud settings on xbox cloud gaming website. Scroll down and toggle use natuve keyboard and mouse option for dayZ & also hide mouse cursor while inactive, refresh the page as requested by better xcloud. Start DayZ - you hqve to use a controller to press A at the start menu (if you dont have a controller rewasd or a different controller emulator can allow you to use a keyboard butron to press A - just close the emulator after you've pressed A) 

After pressing A it will sync accounts & open the Dayz home menu, press f8 and a pop-up comes up saying press f9 to release your mouse, then your mouse and keyboard should work! (if not press esc to exit full screen and then enter fullscreen again and it should recapture your keyboard and mouse) 

Hope that helps!

**Reply by Horsk** [reply]
> tentei no chrome e no edge e não funcionou para mim você pode enviar um vídeo ou uma impressão de como você fez isso


Did you find a solution to this problem?

**Reply by Bdosk** [reply]
> > how did u do that
> 
> You dont need this test script for Day z, i just used it to test on brave as it wouldn't work with the mouse at all in brave.
> 
> How to play on edge with kbm - Install better xcloud extension on microsoft edge via the official website (tapermonkey & better xcloud script) open up the bettwr xcloud settings on xbox cloud gaming website. Scroll down and toggle use natuve keyboard and mouse option for dayZ & also hide mouse cursor while inactive, refresh the page as requested by better xcloud. Start DayZ - you hqve to use a controller to press A at the start menu (if you dont have a controller rewasd or a different controller emulator can allow you to use a keyboard butron to press A - just close the emulator after you've pressed A)
> 
> After pressing A it will sync accounts & open the Dayz home menu, press f8 and a pop-up comes up saying press f9 to release your mouse, then your mouse and keyboard should work! (if not press esc to exit full screen and then enter fullscreen again and it should recapture your keyboard and mouse)
> 
> Hope that helps!

I still can't get it to work. Using ms edge

**Comment by Drigosena** [comment]
![edição-gears-5---game-of-the-year-1736707187641](https://github.com/user-attachments/assets/436bafdb-7ae9-4f94-b4f5-6d54f8012195)
Better Xcloud isn't working with native keyboard and mouse on Gears 5 on my PC.

**Comment by scorpius1025** [comment]
the mouse pointer is not showing in pal world i, am able to click around but , it is like an invisible mouse pointer is there a fix to this ?

**Reply by NamPhanDuc** [reply]
Do you fix that?

**Comment by AlvaroG117** [comment]
I've tested Star wars jedi survivor/9PGC82V0DXFS which is working fine, but the mouse pointer is not showing on the menu.

**Comment by ursusworks** [comment]
Tested Kingdom Come: Deliverance II (9N4W31HSMVVD) in Edge. No pointer in menu, but keyboard works. Full mouse/keyboard in game. WIth KM layout on button icons.

**Reply by ursusworks** [reply]
Seems broken after Title update 1.2.2. Not sure if it's just me or for everyone? I've tested from two different PCs. None work anymore. Stuck at Title screen with no inputs working :(

**Reply by derpgolem** [reply]
Tried several games and they all have the same issue. Seems like others are having this issue as well. Interestingly, I noticed that the "Play with Keyboard and Mouse" section is gone from gamepass and I found similar complaints online as well. Maybe they removed kbm support?

**Comment by Klaedy** [comment]
Avowed doesn't have a menu cursor, but everything else works fine, and for now -in my opinion- the control in action makes up for the occasional absence of a cursor. Thanks.

**Comment by Luc6i** [comment]
Mouse and Keyboard fully supported on Avowed/ "9MSVPJCHHRPR" now with the exception of the left side of skill wheel (no biggie) but everything works

**Reply by Luc6i** [reply]
> Mouse and Keyboard fully supported on Avowed/ "9MSVPJCHHRPR" now with the exception of the left side of skill wheel (no biggie) but everything works

(Update 1.3)

**Comment by ursusworks** [comment]
Tested Atomfall. It's fully working. Mouse in menus and in game. Button layout is also automatically swapped to keyboard

**Comment by ursusworks** [comment]
Tested South of Midnight for fun.

K/M button layout activates.
Full Mouse/Keyboard control in-game
In menus the cursor is hidden. It's easy to start the game, as you can just press "Enter" to continue the game. But it gets frustrating when handling stuff in inventory etc.

**Comment by Crasthiff** [comment]
Tested The First Descendant, working like a charm

**Comment by devilhead97** [comment]
Fully working in Dead by Daylight.

**Comment by frizq010pl** [comment]
How about dead space remake? I can move with keyboard but mouse doesnt work

**Comment by Stuarta99** [comment]
Not sure if I'm doing anything wrong but I've tried going back to MSFS20, which it mentions is supported, but I can't get it to work so not sure if I'm doing something wrong

**Comment by ursusworks** [comment]
VTM: Bloodlines 2 seems to have mouse/keyboard support.
Cursor in menus
Mouse and keyboard works in game. 
Most keys are labeled correctly in the menus, but "E" for action command shows up white. Still works. But no/blank in game label.

**Comment by sha-ir** [comment]
aoe ii working. one-time controller step — set options → interface → input → mouse/keyboard, then menus are mouse-navigable too

**Comment by notalexnot** [comment]
Rainbow Six Siege now added M&K support, would it be possible to add it to the list? I'd be able to test it if necessary. I also ask because it seems to detect the M&K of my computer and doesn't let me change the matchmaking pool to console only. Thanks for the hard work!

**Comment by srikanthamsa** [comment]
tried far cry 6, **working perfectly!** a few issues with micro adjustments while aiming but that's expected because controllers have a built-in "deadzone" area in the center so the camera doesn’t drift.
